plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+# More Viable Plasma
+
+## Introduction
+
+[Minimal Viable Plasma](https://ethresear.ch/t/minimal-viable-plasma/426) (“Plasma MVP”) describes a simple specification for a UTXO-based Plasma chain.
+A key component of the Plasma MVP design is a protocol for “exits,” whereby a user may withdraw back to the root chain any funds available to them on the Plasma chain.
+The protocol presented in the specification requires users sign two signatures for every transaction.
+Concerns over the poor user experience presented by this protocol motivated the search for an alternative exit protocol.
+
+In this document, we describe More Viable Plasma (“MoreVP”), a modification to the Plasma MVP exit protocol that improves user experience.
+The MoreVP exit protocol ensures the security of assets for clients correctly following the protocol.
+We initially present the most basic form of the MoreVP protocol and provide intuition towards its correctness.
+We formalize certain requirements for Plasma MVP exit protocols and provide a proof that this protocol satisfies these requirements in the appendix.
+
+An optimized version of the protocol is presented to account for restrictions of the Ethereum Virtual Machine.
+We further optimize on the observation that the MoreVP exit protocol is only necessary for transactions that are in-flight when a chain becomes byzantine.
+
+We note the existence of certain attack vectors in the protocol, but find that most of these vectors can be largely mitigated and isolated to a relatively small attack surface.
+These attack vectors and their mitigations are described in detail.
+Although we conclude that the design is safe under certain reasonable assumptions about user behavior, some points are highlighted and earmarked for future consideration.
+
+Overall, the MoreVP exit protocol is a significant improvement over the original Plasma MVP exit protocol.
+We can further combine several optimizations to enhance user experience and reduce costs for users.
+Future work will focus on decreasing implementation complexity of the design and minimizing contract gas usage.
+
+
+## Exit Protocol
+
+In this section, we specify the MoreVP exit protocol and give an intuitive argument toward its correctness.
+A formal treatment of the protocol is presented in the appendix.
+
+
+### Definitions
+
+#### Spend Transaction
+
+A spend transaction is any transaction that spends a UTXO present on the Plasma chain.
+
+
+#### Deposit
+
+A deposit creates a new output on the Plasma chain.
```

Comment:
add: ...by depositing in a Plasma contract on the rootchain and the other ways etc...